torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1811,6 +1833,19 @@ def sample_inputs_masked_select(op_info, device, dtype, requires_grad):
            decorators=[_wrap_maybe_warns("floor_divide is deprecated, and will be removed")],
            supports_autograd=False,
            ),
+    OpInfo('linalg.matrix_power',
+           op=torch.linalg.matrix_power,
+           dtypes=floating_and_complex_types(),
+           test_inplace_grad=False,
+           supports_tensor_out=True,
+           decorators=[skipCUDAIfNoMagma, skipCPUIfNoLapack],
```

Comment:
Is matrix inverse working for ROCm? If not it might also need `skipCUDAIfROCm`.
`skipCUDAIfNoMagma` should be `skipCUDAIfNoMagmaAndNoCusolver` because inverse can use either of those.

Reply:
Here are the test durations

```
11.87s call     test/test_ops.py::TestGradientsCUDA::test_fn_gradgrad_linalg_matrix_power_cuda_complex128
5.83s call     test/test_ops.py::TestGradientsCPU::test_fn_gradgrad_linalg_matrix_power_cpu_complex128
3.13s call     test/test_ops.py::TestGradientsCUDA::test_fn_gradgrad_linalg_matrix_power_cuda_float64
2.73s call     test/test_ops.py::TestOpInfoCUDA::test_supported_dtypes_linalg_matrix_power_cuda_complex128
1.79s call     test/test_ops.py::TestGradientsCUDA::test_fn_grad_linalg_matrix_power_cuda_complex128
1.71s call     test/test_ops.py::TestGradientsCPU::test_fn_gradgrad_linalg_matrix_power_cpu_float64
1.36s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_jit_linalg_matrix_power_cuda_complex128
1.34s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_jit_linalg_matrix_power_cuda_complex64
1.24s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_jit_linalg_matrix_power_cuda_float32
1.23s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_jit_linalg_matrix_power_cuda_float64
1.05s call     test/test_ops.py::TestCommonCPU::test_variant_consistency_jit_linalg_matrix_power_cpu_complex128
1.01s call     test/test_ops.py::TestCommonCPU::test_variant_consistency_jit_linalg_matrix_power_cpu_complex64
0.96s call     test/test_ops.py::TestCommonCPU::test_variant_consistency_jit_linalg_matrix_power_cpu_float32
0.95s call     test/test_ops.py::TestCommonCPU::test_variant_consistency_jit_linalg_matrix_power_cpu_float64
0.68s call     test/test_ops.py::TestGradientsCPU::test_fn_grad_linalg_matrix_power_cpu_complex128
0.60s call     test/test_ops.py::TestGradientsCUDA::test_fn_grad_linalg_matrix_power_cuda_float64
0.24s call     test/test_ops.py::TestGradientsCPU::test_fn_grad_linalg_matrix_power_cpu_float64
0.13s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_eager_linalg_matrix_power_cuda_float64
0.13s call     test/test_ops.py::TestOpInfoCPU::test_supported_dtypes_linalg_matrix_power_cpu_complex128
0.13s call     test/test_ops.py::TestCommonCUDA::test_out_linalg_matrix_power_cuda_complex128
0.12s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_eager_linalg_matrix_power_cuda_complex64
0.12s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_eager_linalg_matrix_power_cuda_complex128
0.11s call     test/test_ops.py::TestOpInfoCUDA::test_unsupported_dtypes_linalg_matrix_power_cuda_float16
0.11s call     test/test_ops.py::TestCommonCUDA::test_variant_consistency_eager_linalg_matrix_power_cuda_float32
0.11s call     test/test_ops.py::TestCommonCUDA::test_out_linalg_matrix_power_cuda_complex64
0.11s call     test/test_ops.py::TestOpInfoCUDA::test_unsupported_dtypes_linalg_matrix_power_cuda_bool
0.10s call     test/test_ops.py::TestOpInfoCUDA::test_unsupported_dtypes_linalg_matrix_power_cuda_int64
0.10s call     test/test_ops.py::TestOpInfoCUDA::test_supported_dtypes_linalg_matrix_power_cuda_complex64
0.10s call     test/test_ops.py::TestOpInfoCUDA::test_supported_dtypes_linalg_matrix_power_cuda_float64
0.10s call     test/test_ops.py::TestOpInfoCUDA::test_supported_dtypes_linalg_matrix_power_cuda_float32
0.10s call     test/test_ops.py::TestOpInfoCUDA::test_unsupported_dtypes_linalg_matrix_power_cuda_bfloat16
0.10s call     test/test_ops.py::TestCommonCUDA::test_out_linalg_matrix_power_cuda_float64
0.10s call     test/test_ops.py::TestCommonCUDA::test_out_linalg_matrix_power_cuda_float32
0.10s call     test/test_ops.py::TestOpInfoCUDA::test_unsupported_dtypes_linalg_matrix_power_cuda_int8
0.10s call     test/test_ops.py::TestOpInfoCUDA::test_unsupported_dtypes_linalg_matrix_power_cuda_int32
0.10s call     test/test_ops.py::TestOpInfoCUDA::test_unsupported_dtypes_linalg_matrix_power_cuda_int16
0.10s call     test/test_ops.py::TestOpInfoCUDA::test_unsupported_dtypes_linalg_matrix_power_cuda_uint8
0.05s call     test/test_ops.py::TestGradientsCUDA::test_inplace_gradgrad_linalg_matrix_power_cuda_float64
0.05s call     test/test_ops.py::TestGradientsCUDA::test_inplace_grad_linalg_matrix_power_cuda_complex128
0.05s call     test/test_ops.py::TestGradientsCUDA::test_inplace_grad_linalg_matrix_power_cuda_float64
0.05s call     test/test_ops.py::TestGradientsCUDA::test_inplace_gradgrad_linalg_matrix_power_cuda_complex128
0.01s call     test/test_ops.py::TestCommonCPU::test_variant_consistency_eager_linalg_matrix_power_cpu_complex64
0.01s call     test/test_ops.py::TestCommonCPU::test_variant_consistency_eager_linalg_matrix_power_cpu_complex128
0.01s call     test/test_ops.py::TestCommonCPU::test_variant_consistency_eager_linalg_matrix_power_cpu_float64
0.01s call     test/test_ops.py::TestCommonCPU::test_variant_consistency_eager_linalg_matrix_power_cpu_float32
0.01s call     test/test_ops.py::TestOpInfoCPU::test_unsupported_dtypes_linalg_matrix_power_cpu_int32
0.01s call     test/test_ops.py::TestOpInfoCPU::test_unsupported_dtypes_linalg_matrix_power_cpu_int64
0.01s call     test/test_ops.py::TestOpInfoCPU::test_unsupported_dtypes_linalg_matrix_power_cpu_bool
0.01s call     test/test_ops.py::TestOpInfoCPU::test_unsupported_dtypes_linalg_matrix_power_cpu_uint8
0.01s call     test/test_ops.py::TestOpInfoCPU::test_unsupported_dtypes_linalg_matrix_power_cpu_int16
0.01s call     test/test_ops.py::TestOpInfoCPU::test_unsupported_dtypes_linalg_matrix_power_cpu_int8

(141 durations < 0.005s hidden.  Use -vv to show these durations.)
```